### ISSUE TITLE
Question about training in link prediction？

### ISSUE THREAD

**Issue by maoulee** [apoorvumang/kgt5#15]
Nice work, I have some questions about the training process of KGT5.
(1) 1 vs ALL method: The KGT5 model is trained using the 1 vs ALL method, but the loss function in main_accelerate.py is CE.
(2) Link prediction: I used the provided code to perform link prediction on the WN18RR dataset, and Hit@1 is only 0.108.
Can you provide more details on training and experimental setup?

**Comment by AdrianKs** [comment]
Hi,
(1) KGT5 is trained with a teacher forcing approach using the cross entropy loss. See Appendix B in the paper for more details. Compared to the "standard KGE training" this comes closest to the 1vsAll training method. With 1vsAll+CE we would score the true entity against all other entities. In contrast, with KGT5 we do a similar procedure but on the token-level instead of entity-level.
(2) Unfortunately the code base is not fully prepared yet and the evaluation function in the main branch returns Hits@1 unfiltered only. Depending on the dataset, filtering can have a large impact. For filtered evaluation refer to the description in issue #11 for now. We will merge the other branch into main soon.
